docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+
+```sh
+generated
+└── build
+    └── generated
+        └── ios
+            ├── FBReactNativeSpec
+            │   ├── FBReactNativeSpec-generated.mm
+            │   └── FBReactNativeSpec.h
+            ├── RCTThirdPartyFabricComponentsProvider.h
+            ├── RCTThirdPartyFabricComponentsProvider.mm
+            ├── RTNCalculatorSpec
+            │   ├── RTNCalculatorSpec-generated.mm
+            │   └── RTNCalculatorSpec.h
+            └── react
+                └── renderer
+                    └── components
+                        └── rncore
+                            ├── ComponentDescriptors.h
+                            ├── EventEmitters.cpp
+                            ├── EventEmitters.h
+                            ├── Props.cpp
+                            ├── Props.h
+                            ├── RCTComponentViewHelpers.h
+                            ├── ShadowNodes.cpp
+                            └── ShadowNodes.h
+```
+
+The relevant path for the TurboModule interface is `generated/build/generated/ios/RTNCalculatorSpec`.
+
+See the [CodeGen](./pillars-codegen) section for further details on the generated files.
+
+#### Write the Native iOS Code
+
+Now add the Native code for your TurboModule. Create two files in the `RTNCalculator/ios` folder:
+
+1. The `RTNCalculator.h`, a header file for the module.
+2. The `RTNCalculator.mm`, the implementation of the module.
+
+##### RTNCalculator.h
+
+```objc title="RTNCalculator.h"
+#import <React/RCTBridgeModule.h>
+
+@interface RTNCalculator : NSObject <RCTBridgeModule>
+
+@end
+```
+
+This file defines the interface for the `RTNCalculator` module. Here, we can add any native method we may want to invoke on the view. For this guide, we don't need anything, therefore the interface is empty.
+
+
+##### RTNCalculator.mm
+
+```objc title="RTNCalculator.mm"
+#import "RTNCalculatorSpec.h"
+#import "RTNCalculator.h"
+
+@implementation RTNCalculator
+
+RCT_EXPORT_MODULE(RTNCalculator)
+
+RCT_REMAP_METHOD(add, addA:(NSInteger)a
+                      andB:(NSInteger)b
+                withResolver:(RCTPromiseResolveBlock) resolve
+                withRejecter:(RCTPromiseRejectBlock) reject)
+{
+    NSNumber *result = [[NSNumber alloc] initWithInteger:a+b];
+    resolve(result);
+}
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+    return std::make_shared<facebook::react::NativeCalculatorSpecJSI>(params);
+}
+
+@end
+```
+
+The most important call is to the `RCT_EXPORT_MODULE`, which is required to export the module so that React Native can load the TurboModule.
+
+Then the `RCT_REMAP_METHOD` macro is used to expose the `add` method.
+
+:::info
+There are other macros that can be used to export modules and methods. You view the code that specifies them [here](https://github.com/facebook/react-native/blob/main/React/Base/RCTBridgeModule.h).
+:::
+
+TODO: more description?
```

Comment:
Asking @cipolleschi here for a feedback

Reply:
I would add a small paragraph on the `getTurbomodule:` method:
```
Finally, the `getTurboModule:` is used by the New Architecture to create a proper instance of the TurboModule so that the JavaScript side can properly invoke its methods. The function is defined in (and requested by) the `RTNCalculatorSpec.h` file that is generated by CodeGen.
```
Or something similar.